handlers/anteHandler.go

```diff
@@ -185,3 +188,77 @@ func includeDepositAnteHandler(ctx sdk.Context, ds store.DataStore, msg msgs.Inc
 	}
 	return ctx, sdk.Result{}, false
 }
+
+// validates that the confirmSig msg is valid given the current plasma sidechain state
+func confirmSigAnteHandler(ctx sdk.Context, ds store.DataStore, confirmSigMsg msgs.ConfirmSigMsg, client plasmaConn) (newCtx sdk.Context, res sdk.Result, abort bool) {
+
+	// attempt to recover signers
+	signers := confirmSigMsg.GetSigners()
+	if len(signers) == 0 {
+		return ctx, ErrInvalidTransaction("failed recovering signers").Result(), true
+	}
+
+	expectedSigners := 0
+	if confirmSigMsg.Input1.Signature != [65]byte{} {
+		expectedSigners += 1
+	}
+	if confirmSigMsg.Input2.Signature != [65]byte{} {
+		expectedSigners += 1
+	}
+
+	if len(signers) != expectedSigners {
+		return ctx, ErrInvalidSignature("number of signers does not equal number of signatures").Result(), true
+	}
+
+	res = validateSigMsgInput(ctx, ds, confirmSigMsg.Input1, common.BytesToAddress(signers[0]), client)
+	if !res.IsOK() {
+		return ctx, res, true
+	}
+	if len(signers) > 1 {
+		res = validateSigMsgInput(ctx, ds, confirmSigMsg.Input2, common.BytesToAddress(signers[1]), client)
+		if !res.IsOK() {
+			return ctx, res, true
+		}
+	}
+
+	return ctx, sdk.Result{}, false
+}
+
+// validates the inputs against the output store and returns the amount of the respective input
+func validateSigMsgInput(ctx sdk.Context, ds store.DataStore, input plasma.Input, signer common.Address, client plasmaConn) (sdk.Result) {
```

Comment:
don't need parens around (sdk.Result). also no amount is being returned (comment)